src/core/geometry/element/Element.cpp

```diff
@@ -48,6 +48,21 @@ bool Base::operator!=(const Base& rhs) const {
     return !(*this == rhs);
 }
 
+Base& Base::operator=(const Base& rhs) {
+    if (this == &rhs) {
+        return *this;
+    }
+
+    delete lay_;
+    delete mat_;
```

Comment:
This can not delete layers or materials. It only needs to overwrite the pointer.

Reply:
Directly removed the `=` operator.